### PR TITLE
Mobile - Global styles: Block-specific styles

### LINKED ISSUE #2195: Apply background/foreground colors to blocks
We are almost finished with the [first step of Global Styles support](https://github.com/WordPress/gutenberg/pull/21637). That means we can now inherit color settings from container blocks. But, we aren't yet applying them to the UI.

We need some design point of view to apply background colors. On web, an extra padding is applied when a block gets background color. Should we do a similar thing? @iamthomasbishop If yes, how much padding would you recommend? Also any other thoughts about how it should look on mobile?

Screen Shot 2020-04-27 at 19 36 24

### PR DESCRIPTION
**Pull request by geriux**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2195

This PR is a continuation of https://github.com/WordPress/gutenberg/pull/30684 and https://github.com/WordPress/gutenberg/pull/30810

## Description
This PR adds support to render the theme's block-specific styles. It also adds a reset option in the color picker.

## How has this been tested?

<details><summary>Precondition</summary>
<hr>
To be able to test this feature you need a WordPress 5.8 site running with the Gutenberg plugin and a FSE theme.

- Create a JN testing site with the following options:
  - **Don't** include Jetpack.
  - Include Gutenberg plugin.
  - Include FSE-enabled TT1 Blocks Theme.
- Once the site is created activate the TT1 Blocks Theme.
- Install the WordPress Beta Tester Plugin.
- Activate WordPress Beta Tester.
- Go to Tools > Beta testing.
- Enable the Bleeding edge option.
- Go to Dashboard > Updates.
- Click on Update to the latest nightly.

Now you should have all needed to test this PR.
</p>
</details>

### In the WP Admin dashboard:
- Go to the Site editor.
- To the up right of the page, tap on Global styles
   - Screenshot 2021-06-24 at 12 27 25
- Click on By block type
- Change the default styles (Background color and text color) of different blocks like: `Group`, `Paragraph`, `Heading`, `List`, `Verse`, `Media&Text`, `Columns`, `Column`, `Buttons`.

### On the mobile editor

- Open the WordPress app with metro running from these PRs:
   - `Android` https://github.com/wordpress-mobile/WordPress-Android/pull/14597
   - `iOS` https://github.com/wordpress-mobile/WordPress-iOS/pull/16411
- Log in to your JN site (self-hosted option). 
- Create a new post
- Add the blocks you changed the styles to
- **Expect** to see the new styles in the mobile editor.

### Reset custom color selection

- Add a Paragraph block
- Set a text color
- Expect to see the reset option available
- Tap on reset
- **Expect** to see the default color and not the custom one

## Screenshots
iOS|Android|Reset colors
-|-|-
<kbd></kbd> | <kbd></kbd> | <kbd></kbd>

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by antonis — COMMENTED** [review]
Great work @geriux 👍 
I started testing this on Android and everything looks great till now. The code also LGTM.
<details>
 <summary>Maybe not so great with my color selections :P </summary>

device-2021-06-24-182336
</details>

 I'll resume testing tomorrow with more blocks and the iOS side.

**Review by antonis — APPROVED** [review]
I tested this on Android (Pixel 2 XL/Android 11) and iOS (iPhone SE 2020/iOS 14.4.1) and everything works as expected. The code changes are consistent and overall LGTM 🎉 
Great work @geriux 🙇

**Comment by geriux** [comment]
Hey @ellatrix 👋  I was wondering if you have some time to give this PR a look, I tested the web editor and these changes don't affect the current functionality. 

It adds the `mergedStyle` prop in the shared blocks `List` and `Verse` like we do for `Heading` for example.

Thank you!

File: packages/block-library/src/list/edit.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ export default function ListEdit( {
 	setAttributes,
 	mergeBlocks,
 	onReplace,
+	mergedStyle,
```
Comment (ellatrix) [inline packages/block-library/src/list/edit.js]: Why not simply "style"?

Comment (geriux) [reply packages/block-library/src/list/edit.js]: Thanks for the suggestion @ellatrix and it does make more sense to be named `style`, I renamed these to be more consistent. https://github.com/WordPress/gutenberg/pull/30997/commits/ed6e9d2021b2ffa84bb34988ca7a816e3eae1107

**Comment by ellatrix** [comment]
Ok, for web this seems fine.

**Pull request merged**